### PR TITLE
Wire in focusNode, focusColor, autofocus, and dropdownColor to DropdownButtonFormField

### PR DESCRIPTION
**Pull request by gspencergoog**
## Description

`DropdownButtonFormField` is a wrapper that makes a `DropdownButton` into a form field, but it had some missing parameters that `DropdownButton` had. So I fixed that: this wires through the focus-related attributes, and one that was missing from before (`dropdownColor`).

It also fixes the fact that `DropdownButtonFormField` didn't show itself as focused when it was focused, and I also cleaned up the argument order and fixed some incorrect documentation.

## Related Issues

- No issue filed, but was inspired from this comment: https://github.com/flutter/flutter/issues/42511#issuecomment-612448984

## Tests

- Added tests for testing to make sure that the focus highlight is drawn with the right color, and that the focus color is used.

## Breaking Change

- [X] No, this is *not* a breaking change.

### CODE REVIEW

File: packages/flutter/lib/src/material/dropdown.dart [hunk 1]
```diff
@@ -1426,25 +1430,26 @@ class _DropdownButtonState<T> extends State<DropdownButton<T>> with WidgetsBindi
   }
 }
 
-/// A convenience widget that wraps a [DropdownButton] in a [FormField].
+/// A convenience widget that makes a [DropdownButton] into a [FormField].
 class DropdownButtonFormField<T> extends FormField<T> {
-  /// Creates a [DropdownButton] widget wrapped in an [InputDecorator] and
+  /// Creates a [DropdownButton] widget wrapped in an [InputDecorator] that is a
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/dropdown.dart:1435]: This is a little hard to parse, it sounds like the InputDecorator is a FormField.

File: packages/flutter/lib/src/material/dropdown.dart [hunk 2]
```diff
@@ -1477,32 +1491,46 @@ class DropdownButtonFormField<T> extends FormField<T> {
          autovalidate: autovalidate,
          builder: (FormFieldState<T> field) {
            final _DropdownButtonFormFieldState<T> state = field as _DropdownButtonFormFieldState<T>;
-           final InputDecoration effectiveDecoration = decoration.applyDefaults(
+           final InputDecoration decorationArg =  decoration ?? InputDecoration(focusColor: focusColor);
+           final InputDecoration effectiveDecoration = decorationArg.applyDefaults(
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/dropdown.dart:1495]: I realize that we've been doing this all-along. I wish that InputDecorator computed the defaults.

File: packages/flutter/lib/src/material/dropdown.dart [hunk 3]
```diff
@@ -1512,11 +1540,8 @@ class DropdownButtonFormField<T> extends FormField<T> {
 
   /// The decoration to show around the dropdown button form field.
   ///
-  /// By default, draws a horizontal line under the dropdown button field but can be
-  /// configured to show an icon, label, hint text, and error text.
-  ///
-  /// Specify null to remove the decoration entirely (including the
-  /// extra padding introduced by the decoration to save space for the labels).
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/dropdown.dart:1519]: Don't we need to carry on explaining this detail about the extra padding?

**Review by HansMuller — APPROVED** [review]
LGTM

File: packages/flutter/lib/src/material/dropdown.dart (hunk 3 above)
Comment (gspencergoog) [reply packages/flutter/lib/src/material/dropdown.dart:1519]: Heh. Yeah, that "feature" was never true.  If you specified null, it would assert.

Good point though: I deleted it, meaning to keep the assert, but then I decided that  it was a necessary feature to get the `focusColor` correct in the decorator. I added an appropriate description back in.

File: packages/flutter/lib/src/material/dropdown.dart (hunk 2 above)
Comment (gspencergoog) [reply packages/flutter/lib/src/material/dropdown.dart:1495]: Yeah, and I wish I could access the value of `this.decoration` in a constructor, so I wouldn't have to compute the value twice.

File: packages/flutter/lib/src/material/dropdown.dart (hunk 1 above)
Comment (gspencergoog) [reply packages/flutter/lib/src/material/dropdown.dart:1435]: OK, reworded it.

File: packages/flutter/lib/src/material/dropdown.dart (hunk 2 above)
Comment (gspencergoog) [reply packages/flutter/lib/src/material/dropdown.dart:1495]: Although maybe I misunderstood your comment: did you want a change here?

Comment (HansMuller) [reply packages/flutter/lib/src/material/dropdown.dart:1495]: Just an observation, sorry about the confusion.

**Comment by kamleshwebtech** [comment]
is there any example of solution?
